tests/clickhouse-test

```diff
@@ -1376,40 +1310,118 @@ class TestCase:
 
             if result.status == TestStatus.FAIL:
                 result.description = self.add_info_about_settings(result.description)
+
+            self._cleanup(result.status == TestStatus.OK)
```

Comment:
Should we set the status to `FAIL` if cleanup fails? For example, something is definitely wrong if DROP DATABASE fails with timeout

Reply:
(that's why we did cleanup before checking test status, and that's why `no-drop-if-fail` did not work)

Reply:
>Should we set the status to FAIL if cleanup fails? 

Yes, and the status will be set to `FAIL`, since if `DROP` fails it will throw some exception, either `socket.timeout` or `HTTPError`